src/App.tsx

```diff
@@ -15,48 +16,126 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET,
+  LENGTH,
+  NONE,
+}
+
+type ReorderedOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getOrderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderedOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return a.localeCompare(b);
+
+      case SortType.LENGTH:
+        return a.length - b.length;
+
+      default:
+        return SortType.NONE;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReverse] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const reverse = () => {
+    setReverse(current => !current);
+  };
+
+  const reset = () => {
+    setReverse(false);
+    setSortType(SortType.NONE);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const sortByAlphabet = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const getReorderedGoods = getOrderedGoods(
```

Comment:
It's just a recommendation, not a change request. Read about useMemo hook.
In a nutshell, now you call getOrderedGoods on every render of the component, which may cause laggy experience on slow devices. With help of useMemo hook you could avoid this problem.
https://beta.reactjs.org/apis/react/useMemo